onnx/reference/custom_element_types.py

```diff
@@ -5,10 +5,72 @@
 
 import numpy as np
 
+try:
+    import ml_dtypes
+except ImportError:
+    ml_dtypes = None
+
 bfloat16 = np.dtype((np.uint16, {"bfloat16": (np.uint16, 0)}))
```

Comment:
I would recommend using ml_dtypes to replace the custom dtypes. This way we get both representational correctness (bc compatible) and enable easy computation on these types.